lib/index.js

```diff
@@ -29,214 +29,225 @@ function resolveInputDirectory(inputNode) {
+        if (resultSeverity === 1 && throwOnWarn) {
+          throw new Error('rules violation with `warn` severity level');
+        }
+
+        if (resultSeverity >= 2 && (throwOnWarn || throwOnError)) {
+          throw new Error('rules violation with `error` severity level');
+        }
       }
     }
-  }
-
-  return { output: results.output };
-};
 
-/**
- * Factory function that allows to return something different
- * instead of a single EslintValidationFilter.
- *
- * @experimental
- */
-EslintValidationFilter.create = function(inputNode, _options) {
-  let options = Object.assign({}, _options, {
-    [FACTORY_METHOD_USED]: true
-  });
-
-  if (!options.group) {
-    return new EslintValidationFilter(inputNode, options);
+    return { output: results.output };
   }
 
-  if (typeof options.testGenerator !== 'string' || testGeneratorNames.indexOf(options.testGenerator) === -1) {
-    throw new Error(`The "group" options can only be used with a "testGenerator" option of: ${testGeneratorNames}`);
-  }
+  /**
+   * Factory function that allows to return something different
+   * instead of a single EslintValidationFilter.
+   *
+   * @experimental
+   */
+  static create(inputNode, _options) {
+    let options = Object.assign({}, _options, {
+      [FACTORY_METHOD_USED]: true,
+    });
+
+    if (!options.group) {
+      return new EslintValidationFilter(inputNode, options);
+    }
 
-  let testGenerator = testGenerators[options.testGenerator];
+    if (
+      typeof options.testGenerator !== 'string' ||
+      testGeneratorNames.indexOf(options.testGenerator) === -1
+    ) {
+      throw new Error(
+        `The "group" options can only be used with a "testGenerator" option of: ${
+          testGeneratorNames
+        }`
+      );
+    }
 
-  let header = testGenerator.header(options.group);
-  let footer = testGenerator.footer();
+    let testGenerator = testGenerators[options.testGenerator];
 
-  let eslint = new EslintValidationFilter(inputNode, options);
+    let header = testGenerator.header(options.group);
+    let footer = testGenerator.footer();
 
-  return concat(eslint, {
-    outputFile: `/${options.group}.lint-test.js`,
-    header,
-    inputFiles: ['**/*.lint-test.js'],
-    footer,
-    sourceMapConfig: { enabled: false },
-    allowNone: true,
-  });
-};
+    let eslint = new EslintValidationFilter(inputNode, options);
 
-Object.defineProperty(EslintValidationFilter, 'testGenerators', {
-  get() {
+    return concat(eslint, {
+      outputFile: `/${options.group}.lint-test.js`,
+      header,
+      inputFiles: ['**/*.lint-test.js'],
+      footer,
+      sourceMapConfig: { enabled: false },
+      allowNone: true,
+    });
+  }
+
+  static get testGenerators() {
     return testGeneratorNames.slice(0);
   }
+}
+
+// This is so that we remain backwards compatibile. Some people may invoke
+// without the `new` keyword (which is required for ES6 classes).
+const EslintValidationFilter = new Proxy(_EslintValidationFilter, {
```

Comment:
I don’t think we can rely on Proxy being available (would have to double check what Node 4 has to confirm)...

Reply:
I think you can get the same general result with a “normal” function here:

```
function EslintValidationFilter(inputNode, options) {
  if (this !== EslintValidationFilter) {
    console.warn(‘invoking broccoli-lint-eslint without new will be removed in a future version’);
  }

  return new _ESlintValidationFilter(inputNode, options);
}
```

Reply:
I believe this snippet (or a variant) will allow us to migrate to classes internally and keep newless Support...